lib/active_model_serializers/adapter/json_api/meta.rb

```diff
@@ -18,7 +18,7 @@ def initialize(serializer)
 
           # Use the return value of the block unless it is nil.
           if serializer._meta.respond_to?(:call)
-            @value = instance_eval(&serializer._meta)
```

Comment:
Not sure how it would benefit from inheriting from `Field`. But I can definitely see a common behavior that could be reuse with:
* the block `instance_exec`
* the `:if`, `:unless` conditionals

I guess I could try something out in a different PR and see if I can come up with something clean.